### PR TITLE
Form config test for military service page

### PR DESCRIPTION
**Pull request by raquelromano-gov**
Hi Jeff,
This isn't ready to review, but wondering if you can unblock me. I can't seem to be able to figure out track down this missing reference to #/definitions/dateRange

Would this be something in the vets-json-schema repo that needs to be updated? I'm not clear on how/whether the json files in dist/ are kept in sync with the definitions.

Thanks in advance for any help!
Raquel

```
1) Edu 1995 militaryService servicePeriods should render:
     Error: Could not find a definition for #/definitions/dateRange.
      at findSchemaDefinition (node_modules/react-jsonschema-form/lib/utils.js:368:9)
      at retrieveSchema (node_modules/react-jsonschema-form/lib/utils.js:379:20)
      at toIdSchema (node_modules/react-jsonschema-form/lib/utils.js:480:19)
      at toIdSchema (node_modules/react-jsonschema-form/lib/utils.js:492:22)
      at toIdSchema (node_modules/react-jsonschema-form/lib/utils.js:484:12)
      at toIdSchema (node_modules/react-jsonschema-form/lib/utils.js:492:22)
      at Form.getStateFromProps (node_modules/react-jsonschema-form/lib/components/Form.js:131:44)
      at new Form (node_modules/react-jsonschema-form/lib/components/Form.js:102:25)
      at node_modules/react/lib/ReactCompositeComponent.js:294:18
      at measureLifeCyclePerf (node_modules/react/lib/ReactCompositeComponent.js:74:12)
      at [object Object].ReactCompositeComponentMixin._constructComponentWithoutOwner (node_modules/react/lib/ReactCompositeComponent.js:293:16)
      at [object Object].ReactCompositeComponentMixin._constructComponent (node_modules/react/lib/ReactCompositeComponent.js:279:21)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:187:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:47:35)
      at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (node_modules/react/lib/ReactMultiChild.js:240:44)
      at ReactDOMComponent.Mixin._createInitialChildren (node_modules/react/lib/ReactDOMComponent.js:699:32)
      at ReactDOMComponent.Mixin.mountComponent (node_modules/react/lib/ReactDOMComponent.js:524:12)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:47:35)
      at [object Object].ReactCompositeComponentMixin.performInitialMount (node_modules/react/lib/ReactCompositeComponent.js:370:34)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:257:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:47:35)
      at [object Object].ReactCompositeComponentMixin.performInitialMount (node_modules/react/lib/ReactCompositeComponent.js:370:34)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:257:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:47:35)
      at [object Object].ReactCompositeComponentMixin.performInitialMount (node_modules/react/lib/ReactCompositeComponent.js:370:34)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:257:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:47:35)
      at mountComponentIntoNode (node_modules/react/lib/ReactMount.js:105:32)
      at ReactReconcileTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:138:20)
      at batchedMountComponentIntoNode (node_modules/react/lib/ReactMount.js:127:15)
      at ReactDefaultBatchingStrategyTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:138:20)
      at Object.ReactDefaultBatchingStrategy.batchedUpdates (node_modules/react/lib/ReactDefaultBatchingStrategy.js:63:19)
      at Object.batchedUpdates (node_modules/react/lib/ReactUpdates.js:98:20)
      at Object.ReactMount._renderNewRootComponent (node_modules/react/lib/ReactMount.js:321:18)
      at Object.ReactMount._renderSubtreeIntoContainer (node_modules/react/lib/ReactMount.js:402:32)
      at Object.ReactMount.render (node_modules/react/lib/ReactMount.js:423:23)
      at Object.ReactTestUtils.renderIntoDocument (node_modules/react/lib/ReactTestUtils.js:84:21)
      at Context.<anonymous> (test/edu-benefits/1995/config/militaryService.unit.spec.jsx:14:33)
```

### CODE REVIEW

**Comment by jbalboni** [comment]
@raquelromano-gov That's happening because a field in the schema for the service periods page has a `$ref` pointing to dateRange, but that definition isn't included with the schema, which is due to the way I set up the `DefinitionTester` component. In the actual schemaform code, the definitions that are in `defaultDefinitions` in the form config are added to each page's schema in the Redux [reducer](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/common/schemaform/reducers/index.js#L51) when everything first loads. `DefinitionTester` is meant to run the same startup code that the Redux reducer does, but it doesn't do this.

The quick fix for this is to add the definitions to your schema:

```js
const { schema, uiSchema } = formConfig.chapters.militaryService.pages.servicePeriods;
schema.definitions = formConfig.defaultDefinitions;
```

A better fix is probably to add `formConfig` as an (optional) prop for `DefinitionTester` and add the definitions to the schema in that component's constructor.

**Comment by raquelromano-gov** [comment]
Thanks! Will take a stab at fixing DefinitionTester

File: test/util/schemaform-utils.jsx [hunk 1]
```diff
@@ -15,6 +15,9 @@ export class DefinitionTester extends React.Component {
     const uiSchema = props.uiSchema;
     const data = props.data;
     let schema = updateRequiredFields(props.schema, uiSchema, data, state);
+    if (props.definitions) {
```
Comment (jbalboni) [inline test/util/schemaform-utils.jsx]: It's also possible that a specific page would have definitions already in its schema, so we probably want to merge the two like in the reducer code:

`schema = _.assign({ definitions: props.definitions }, schema);`

File: test/util/schemaform-utils.jsx [hunk 2]
```diff
@@ -15,6 +15,9 @@ export class DefinitionTester extends React.Component {
...
+      schema.definitions = props.definitions;
```
Comment (jbalboni) [inline test/util/schemaform-utils.jsx]: It's also possible that a specific page would have definitions already in its schema, so we probably want to merge the two like in the reducer code:

`schema = _.assign({ definitions: props.definitions }, schema);`

**Comment by jbalboni** [comment]
@raquelromano-gov Let me know when you want me to take another look at this!

**Comment by raquelromano-gov** [comment]
Oops---I had forgotten to submit his comment:

Thanks, still need to test for hidden fields on this page, but can you review the DefinitionTester changes first so they're in place?

**Comment by jbalboni** [comment]
@raquelromano-gov I made one comment on it, looks good to me otherwise.

File: test/util/schemaform-utils.jsx (hunk 2 above)
Comment (raquelromano-gov) [inline test/util/schemaform-utils.jsx]: Ah yes..done.

Comment (raquelromano-gov) [inline test/util/schemaform-utils.jsx]: Actually, if we're merging, is there no need to check whether props.definitions is not null?

Comment (raquelromano-gov) [inline test/util/schemaform-utils.jsx]: Confession: am new to lodash but browsed the docs and changed this to:
```
 schema = _.merge(schema, { definitions: props.definitions });
```

File: test/util/schemaform-utils.jsx [hunk 3]
```diff
@@ -15,6 +17,7 @@ export class DefinitionTester extends React.Component {
     const uiSchema = props.uiSchema;
     const data = props.data;
     let schema = updateRequiredFields(props.schema, uiSchema, data, state);
+    schema = _.merge(schema, { definitions: props.definitions });
```
Comment (jbalboni) [inline test/util/schemaform-utils.jsx]: Should be the other way around, I think: `_.merge({ definitions: props.definitions }, schema);`. That way the definitions in `schema` take precedence.

**Pull request merged**